### PR TITLE
Explicitly track debuggee thread state to detect thread exit/debug event races

### PR DESCRIPTION
**Pull request by ranweiler**
- Define an enum to track the debugger's current understanding of debuggee thread state
- Update our private suspend/resume methods to update and return the current state
- Detect thread exit/debug event races in suspend/resume calls

The win32 `SuspendThread()`/`ResumeThread()` calls increment/decrement a per-thread suspend counter. When the counter is 0, the thread is runnable, else it is suspended.

The win32 functions return the _previous_ suspend count, and `-1` on error. The full set of error descriptions is not part of the public API, but can definitely be caused by a lack of [thread access rights](https://docs.microsoft.com/en-us/windows/win32/procthread/thread-security-and-access-rights), or an exited thread.

We always have the `THREAD_SUSPEND_RESUME` access right for any `ThreadInfo` thread, because we observe it as a debugger of the thread's process. If we get an "access denied" error, then this means that the thread has exited during our current debugger loop operation, but our debugger has not _yet_ been delivered the relevant debug event.

This PR does not _fix_ our handling of racy thread exits, but updates our modeling of debuggee threads to _enable_ robust handling of threads when performing multi-threaded single-stepping.

### CODE REVIEW

File: src/agent/debugger/src/target.rs [hunk 1]
```diff
@@ -31,55 +31,88 @@ pub(crate) enum StepState {
     SingleStep,
 }
 
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+enum ThreadState {
+    Running,
```
Comment (ranweiler) [inline src/agent/debugger/src/target.rs:36]: This is misleading; rename to `Runnable`.

**Review by stishkin — APPROVED** [review]

**Review by bmc-msft — APPROVED** [review]

**Comment by ranweiler** [comment]
Manually testing.

**Comment by ranweiler** [comment]
Manually tested coverage recording and SARIF generation.

**Comment by ranweiler** [comment]
Manual testing complete.

**Pull request merged**